__tests__/app.test.js

```diff
@@ -174,3 +174,54 @@ describe('/api/articles', () => {
         })
     })
 })
+
+describe('/api/articles/:article_id/comments', () => {
+    test('POST 201 - post a comment to an article', () => {
+        const newComment = { username: "icellusedkars", body: "This article is great!" }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(newComment)
+        .expect(201)
+        .then(({body}) => {
+            const { comment } = body
+            expect(comment).toMatchObject({
+                comment_id: expect.any(Number),
+                article_id: 2,
+                author: "icellusedkars",
+                body: "This article is great!",
+                votes: 0,
+                created_at: expect.any(String)
+            })
+        })
+    })
+    test('POST 400 - PSQL error: comment malformed', () => {
+        const newComment = { body: "This article is great!" }
+        return request(app)
+        .post('/api/articles/banana/comments')
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad request')
+        })
+    })
+    test('POST 400 - PSQL error: failed schema validation', () => {
+        const newComment = { username: 4, body: "This article is great!" }
```

Comment:
We would expect a 404 in this instance if the username doesn't exist 👍

Reply:
Looking at the seed, it doesn't seem I can fail the schema validation on the comment body because username/author and body will work with VARCHAR, I've done this error on INTs before the other way round. I don't want to get too deep into user validation on this ticket so I've taken out message with the username and just done a malformed (missing username) as PSQL error and then the 'banana path'. I've kept the 404 for the article_id = 99999. Hope that makes sense!